internal/pipeline/types.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// This file defines the type structures used by pipeline
+package pipeline
+
+import (
+	"sigs.k8s.io/kustomize/kyaml/yaml"
+)
+
+// Function defines an item in the pipeline function list
+type Function struct {
+	// Image is the image name that a function will use
+	Image string `yaml:"image,omitempty"`
+	// Config specifies an inline k8s resource used as the function config.
+	// Config, ConfigPath, and ConfigMap fields are mutually exclusive.
+	Config yaml.Node `yaml:"config,omitempty"`
```

Comment:
This is interesting. Config, ConfigPath and ConfigMap have One-Of (union) semantics so only one of them needs to be set. They will be best represented if pointers are used to refer to them. But in this case, ConfigPath being string is ok because zero value will help here to indicate absence of it. ConfigMap is already a pointer as it is a map. Config should point to *yaml.Node ?